### ISSUE TITLE
azure-cli package repo using old signing key

### ISSUE THREAD

**Issue by sptramer** [Azure/azure-cli#7833]
**Describe the bug**

It appears that the 'azure-cli' xenial release repo is still using the old signing key (short key `EB3E94ADBE1229CF`) instead of the appropriate new key (full key `52E16F86FEE04B979B07E28DB02C46DF417A0893`).

The package itself is properly signed, but this prevents using `https://packages.microsoft.com` as a keyserver.

**To Reproduce**

Ensure that the azure-cli repo and the GPG signing key are not installed on the machine. The best way to do this is through a clean Docker image of xenial. Within a shell session with the `ubuntu:xenial` container:

```bash
apt-get update
apt-get install apt-transport-https lsb-release software-properties-common -y
apt-add-repository "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $(lsb_release -cs) main" \
        --keyserver https://packages.microsoft.com \
        --update
apt-get update
```

Output from the update command includes:

```
W: GPG error: https://packages.microsoft.com/repos/azure-cli xenial InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY EB3E94ADBE1229CF
W: The repository 'https://packages.microsoft.com/repos/azure-cli xenial InRelease' is not signed.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.
```

**Expected behavior**

The signing key for the repository matches the signing key for the package, and there is no error from the keyserver.

**Environment summary**

* Ubuntu Xenial 16.04

**Additional context**

This may affect other Ubuntu releases, or other packages signed with the same key but using other distributions.

**Comment by sptramer** [comment]
@tjprescott - It _may_ be the case that the CLI documentation is old instead of there being a build configuration issue (this key was taken from the proxy troubleshooting section.) GPG reports the key from Microsoft as `BC528686B50D79E339D3721CEB3E94ADBE1229CF`.

**Comment by mud5150** [comment]
Someone should update the documentation. That key worked.

**Comment by tjprescott** [comment]
@sptramer which documentation needs to be updated?

**Comment by sptramer** [comment]
@mud5150 Please tell me which part of the apt install documentation needs to be updated.

@tjprescott Can you double-check the key signing on your side? I think that it's working now - the apt install instructions now use the keyserver (https://docs.microsoft.com/en-us/cli/azure/install-azure-cli-apt?view=azure-cli-latest)

**Comment by mud5150** [comment]
Ahh I see. It does look like the key in the doc is correct. I went in some circles because I am running into the  `gpg dirmngr IPC connect call failed` on WSL. At some point I ended up looking at another github issue that contained the old key and kept using that from my command history. My bad.

**Comment by tjprescott** [comment]
I believe then that this issue can be closed. Please reopen if you feel that's not the case.

**Comment by ruifengma** [comment]
I have this issue while installing the CUDA driver. When execute `sudo apt-get update` and I got this error
`Hit:1 http://azure.archive.ubuntu.com/ubuntu xenial InRelease
Hit:2 http://azure.archive.ubuntu.com/ubuntu xenial-updates InRelease
Hit:3 http://azure.archive.ubuntu.com/ubuntu xenial-backports InRelease
Hit:4 http://security.ubuntu.com/ubuntu xenial-security InRelease
Get:5 https://packages.microsoft.com/repos/azure-cli xenial InRelease [3188 B]
Ign:6 http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64  InRelease
Hit:7 http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64  Release
Ign:5 https://packages.microsoft.com/repos/azure-cli xenial InRelease
Fetched 3188 B in 1s (2721 B/s)
Reading package lists... Done
W: GPG error: https://packages.microsoft.com/repos/azure-cli xenial InRelease: The following signatures couldn't be verified because the public key is not available: NO_PUBKEY EB3E94ADBE1229CF
W: The repository 'https://packages.microsoft.com/repos/azure-cli xenial InRelease' is not signed.
N: Data from such a repository can't be authenticated and is therefore potentially dangerous to use.
N: See apt-secure(8) manpage for repository creation and user configuration details.`
